lib/validator/validator.go

```diff
@@ -301,6 +302,21 @@ func validatePoolMetadata(v *validator.Validate, structLevel *validator.StructLe
 	}
 }
 
+func validateIPIPConfiguration(v *validator.Validate, structLevel *validator.StructLevel) {
+	ipip := structLevel.CurrentStruct.Interface().(api.IPIPConfiguration)
+
+	log.Debugf("IPIP: Enabled %b, Mode: %s", ipip.Enabled, ipip.Mode)
```

Comment:
This could actually be done with a field level validator (see examples that just use a regex for checking values) at the top of the file.